src/deprecated.jl

```diff
@@ -56,4 +56,19 @@ for T in (:Thunk, :InplaceableThunk)
 end
 
 
-Base.@deprecate InplaceableThunk(t::Thunk, add!) InplaceableThunk(add!, t)
\ No newline at end of file
+Base.@deprecate InplaceableThunk(t::Thunk, add!) InplaceableThunk(add!, t)
+
+const CONVERT_DEPRECATION = "convert(::Type{T}, t::Tangent) is deprecated, use ChainRulesCore.backing(t) instead"
+
+function Base.convert(::Type{<:NamedTuple}, comp::Tangent{<:Any, <:NamedTuple})
```

Comment:
Annoyingly it's not possible to `@deprecate Base.convert...`, i.e. adding the `Base` prefix gives the following error:
```julia
ERROR: LoadError: LoadError: LoadError: invalid usage of @deprecate
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] var"@deprecate"(__source__::LineNumberNode, __module__::Module, old::Any, new::Any, ex::Any)
    @ Base ./deprecated.jl:62
  [3] var"@deprecate"(__source__::LineNumberNode, __module__::Module, old::Any, new::Any)
    @ Base ./deprecated.jl:39
  [4] include(mod::Module, _path::String)
    @ Base ./Base.jl:386
  [5] include(x::String)
    @ ChainRulesCore ~/JuliaEnvs/ChainRulesCore.jl/src/ChainRulesCore.jl:1
  [6] top-level scope
    @ ~/JuliaEnvs/ChainRulesCore.jl/src/ChainRulesCore.jl:35
  [7] include
    @ ./Base.jl:386 [inlined]
  [8] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1235
  [9] top-level scope
    @ none:1
 [10] eval
    @ ./boot.jl:360 [inlined]
 [11] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [12] top-level scope
    @ none:1
```

Reply:
To deprecate something from another module you need to do `import Base: convert`
`@deprecate` doesn't support qualified names

Reply:
I've tried this but kept `Base.convert` 🤦